packages/headless/src/controllers/commerce/core/parameter-manager/headless-core-parameter-manager.test.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  CommerceSearchParameters,
+  restoreSearchParameters,
+} from '../../../../features/commerce/search-parameters/search-parameter-actions';
+import {searchParametersDefinition} from '../../../../features/commerce/search-parameters/search-parameter-schema';
+import {executeSearch} from '../../../../features/commerce/search/search-actions';
+import {buildMockCommerceState} from '../../../../test/mock-commerce-state';
+import {
+  buildMockCommerceEngine,
+  MockedCommerceEngine,
+} from '../../../../test/mock-engine-v2';
+import {
+  buildCoreParameterManager,
+  ParameterManager,
+} from './headless-core-parameter-manager';
+
+jest.mock(
+  '../../../../features/commerce/search-parameters/search-parameter-actions'
+);
+jest.mock('../../../../features/commerce/search/search-actions');
+
+describe('product listing parameter manager', () => {
+  let engine: MockedCommerceEngine;
+  let parameterManager: ParameterManager<CommerceSearchParameters>;
+
+  function initEngine(preloadedState = buildMockCommerceState()) {
+    engine = buildMockCommerceEngine(preloadedState);
+  }
+
+  function initParameterManager(props: Partial<CommerceSearchParameters> = {}) {
+    parameterManager = buildCoreParameterManager(engine, {
+      initialState: {parameters: {}},
+      parametersDefinition: searchParametersDefinition,
+      activeParametersSelector: (_) => ({q: ''}),
+      restoreActionCreator: restoreSearchParameters,
+      fetchResultsActionCreator: executeSearch,
+      enrichParameters: (_, activeParams) => ({
+        q: activeParams.q!,
+      }),
+      ...props,
+    });
+  }
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    initEngine();
+    initParameterManager();
+  });
+
+  it('exposes #subscribe method', () => {
+    expect(parameterManager.subscribe).toBeTruthy();
+  });
+
+  describe('initialization', () => {
+    it('validates initial state against schema', () => {});
+
+    it('dispatches #restoreActionCreator', () => {
```

Comment:
Feel free to apply or not, but you could also test with a non empty initial state.